lxd/instance/drivers/driver_lxc.go

```diff
@@ -1885,20 +1935,22 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 		return "", nil, errors.Wrap(err, "Set last ID map")
 	}
 
+	idmapType := d.IdmappedStorage(d.RootfsPath())
+
 	// Once mounted, check if filesystem needs shifting.
-	if !nextIdmap.Equals(diskIdmap) && !(diskIdmap == nil && d.state.OS.Shiftfs) {
+	if !nextIdmap.Equals(diskIdmap) && !(diskIdmap == nil && idmapType != idmap.IdmapStorageNone) {
+		storageType, err := d.getStorageType()
+		if err != nil {
+			return "", nil, errors.Wrap(err, "Storage type")
+		}
```

Comment:
Why is this being moved?

Reply:
This was left-over when it was still considered that the storagetype could be used to make inferences or be recorded in the storage map.